src/main/java/hudson/plugins/favorite/Favorites.java

```diff
@@ -136,6 +136,10 @@ public boolean hasNext() {
                     public Item next() {
                         return jenkins.getItemByFullName(iterator.next());
                     }
+
+                    /* Java 7 compatibility fallback */
+                    @Override
+                    public void remove() { throw new UnsupportedOperationException(); }
```

Comment:
The built failed because of that. And after some googling it seems that depending on the JDK that builds the plugin this is needed. When built with JDK 8 it is not necessary.
When building with JDK 7 and lesser it is. So either fix the jenkins or keep this :)